File: ClinicaVeterinaria/logica/Ticket.cs
using System;

namespace ClinicaVeterinaria.logica
{
    public class Ticket
    {
        public int intNumero { get; set; }
        public string strMotivo { get; set; }
        public string strSintomas { get; set; }
        public string strTipoConsulta { get; set; }
        public string strPrioridad { get; set; }
        public string strEstado { get; set; }
        public DateTime dtFechaCreacion { get; set; }
        public string strDiagnostico { get; set; }
        public string strTratamiento { get; set; }
        public bool blnEscalado { get; set; }
        public Mascota objMascota { get; set; }
        public Veterinario objVeterinarioAsignado { get; set; }
        public Bitacora objBitacora { get; set; }

        public Ticket(int intNumero, string strMotivo, string strSintomas, string strTipoConsulta, string strPrioridad, Mascota objMascota)
        {
            this.intNumero = intNumero;
            this.strMotivo = strMotivo;
            this.strSintomas = strSintomas;
            this.strTipoConsulta = strTipoConsulta;
            this.strPrioridad = strPrioridad;
            this.strEstado = "Abierto";
            this.dtFechaCreacion = DateTime.Now;
            this.blnEscalado = false;
            this.objMascota = objMascota;
            this.objBitacora = new Bitacora();
            objBitacora.registrarEvento($"Ticket creado para la mascota {objMascota.strNombre} (motivo: {strMotivo})");
        }

        public void asignar(Veterinario objVeterinario)
        {
            if (objVeterinarioAsignado != null)
            {
                objVeterinarioAsignado.liberarCarga();
                objBitacora.registrarEvento($"Reasignado de {objVeterinarioAsignado.strNombre} a {objVeterinario.strNombre}");
            }
            else
            {
                objBitacora.registrarEvento($"Asignado al veterinario {objVeterinario.strNombre}");
            }
            objVeterinarioAsignado = objVeterinario;
            objVeterinario.aumentarCarga();
            strEstado = "Asignado";
        }

        public void registrarHallazgo(string strTipo, string strDescripcion, string strGravedad)
        {
            objBitacora.registrarEvento($"Hallazgo [{strTipo}] ({strGravedad}): {strDescripcion}");
        }

        public void resolver(string strDiagnostico, string strTratamiento)
        {
            this.strDiagnostico = strDiagnostico;
            this.strTratamiento = strTratamiento;
            strEstado = "Resuelto";
            objBitacora.registrarEvento($"Consulta resuelta - Diagnostico: {strDiagnostico} | Tratamiento: {strTratamiento}");
        }

        public void cerrar()
        {
            strEstado = "Cerrado";
            objVeterinarioAsignado?.liberarCarga();
            objBitacora.registrarEvento("Ticket cerrado");
        }

        public void escalar(string strMotivo, Veterinario objVeterinarioSenior)
        {
            string strPrioridadAnterior = strPrioridad;
            strPrioridad = "Emergencia";
            blnEscalado = true;
            strEstado = "Escalado";
            objBitacora.registrarEvento($"Escalado ({strPrioridadAnterior} -> {strPrioridad}): {strMotivo}");
            if (objVeterinarioSenior != null && objVeterinarioSenior != objVeterinarioAsignado)
            {
                asignar(objVeterinarioSenior);
                strEstado = "Escalado";
            }
        }

        public void mostrarResumen()
        {
            Console.WriteLine($"Ticket #{intNumero} - {strMotivo}");
            Console.WriteLine($"  Mascota: {objMascota.strNombre} ({objMascota.strEspecie}) - Dueno: {objMascota.objDueno.strNombre}");
            Console.WriteLine($"  Tipo de consulta: {strTipoConsulta} | Prioridad: {strPrioridad} | Estado: {strEstado}");
            Console.WriteLine($"  Sintomas: {strSintomas}");
            Console.WriteLine($"  Fecha de creacion: {dtFechaCreacion:dd/MM/yyyy HH:mm}");
            Console.WriteLine($"  Veterinario asignado: {(objVeterinarioAsignado != null ? objVeterinarioAsignado.strNombre : "Sin asignar")}");
            if (!string.IsNullOrEmpty(strDiagnostico))
                Console.WriteLine($"  Diagnostico: {strDiagnostico} | Tratamiento: {strTratamiento}");
            Console.WriteLine($"  Escalado: {(blnEscalado ? "Si" : "No")}");
        }

        public void mostrarBitacora()
        {
            mostrarResumen();
            objBitacora.mostrarBitacora();
        }
    }
}

File: BlueJ_Proyecto/Ticket.java
import java.util.Date;

/**
 * Caso clinico (ticket) abierto por un Cliente para una Mascota.
 */
public class Ticket
{
    private int numero;
    private String motivo;
    private String sintomas;
    private String tipoConsulta;
    private String prioridad;
    private String estado;
    private Date fechaCreacion;
    private String diagnostico;
    private String tratamiento;
    private boolean escalado;
    private Mascota mascota;
    private Veterinario veterinarioAsignado;
    private Bitacora bitacora;

    public Ticket(int numero, String motivo, String sintomas, String tipoConsulta, String prioridad, Mascota mascota)
    {
        this.numero = numero;
        this.motivo = motivo;
        this.sintomas = sintomas;
        this.tipoConsulta = tipoConsulta;
        this.prioridad = prioridad;
        this.estado = "Abierto";
        this.fechaCreacion = new Date();
        this.escalado = false;
        this.mascota = mascota;
        this.bitacora = new Bitacora();
        bitacora.registrarEvento("Ticket creado para la mascota " + mascota.getNombre() + " (motivo: " + motivo + ")");
    }

    public void asignar(Veterinario veterinario)
    {
        if (veterinarioAsignado != null)
        {
            veterinarioAsignado.liberarCarga();
            bitacora.registrarEvento("Reasignado de " + veterinarioAsignado.getNombre() + " a " + veterinario.getNombre());
        }
        else
        {
            bitacora.registrarEvento("Asignado al veterinario " + veterinario.getNombre());
        }
        veterinarioAsignado = veterinario;
        veterinario.aumentarCarga();
        estado = "Asignado";
    }

    public void registrarHallazgo(String tipo, String descripcion, String gravedad)
    {
        bitacora.registrarEvento("Hallazgo [" + tipo + "] (" + gravedad + "): " + descripcion);
    }

    public void resolver(String diagnostico, String tratamiento)
    {
        this.diagnostico = diagnostico;
        this.tratamiento = tratamiento;
        estado = "Resuelto";
        bitacora.registrarEvento("Consulta resuelta - Diagnostico: " + diagnostico + " | Tratamiento: " + tratamiento);
    }

    public void cerrar()
    {
        estado = "Cerrado";
        if (veterinarioAsignado != null)
        {
            veterinarioAsignado.liberarCarga();
        }
        bitacora.registrarEvento("Ticket cerrado");
    }

    public void escalar(String motivo, Veterinario veterinarioSenior)
    {
        String prioridadAnterior = prioridad;
        prioridad = "Emergencia";
        escalado = true;
        estado = "Escalado";
        bitacora.registrarEvento("Escalado (" + prioridadAnterior + " -> " + prioridad + "): " + motivo);
        if (veterinarioSenior != null && veterinarioSenior != veterinarioAsignado)
        {
            asignar(veterinarioSenior);
            estado = "Escalado";
        }
    }

    public int getNumero()
    {
        return numero;
    }

    public String getMotivo()
    {
        return motivo;
    }

    public String getSintomas()
    {
        return sintomas;
    }

    public String getTipoConsulta()
    {
        return tipoConsulta;
    }

    public String getPrioridad()
    {
        return prioridad;
    }

    public String getEstado()
    {
        return estado;
    }

    public Date getFechaCreacion()
    {
        return fechaCreacion;
    }

    public String getDiagnostico()
    {
        return diagnostico;
    }

    public String getTratamiento()
    {
        return tratamiento;
    }

    public boolean isEscalado()
    {
        return escalado;
    }

    public Mascota getMascota()
    {
        return mascota;
    }

    public Veterinario getVeterinarioAsignado()
    {
        return veterinarioAsignado;
    }

    public Bitacora getBitacora()
    {
        return bitacora;
    }

    public void mostrarResumen()
    {
        System.out.println("Ticket #" + numero + " - " + motivo);
        System.out.println("  Mascota: " + mascota.getNombre() + " (" + mascota.getEspecie() + ") - Dueno: " + mascota.getDueno().getNombre());
        System.out.println("  Tipo de consulta: " + tipoConsulta + " | Prioridad: " + prioridad + " | Estado: " + estado);
        System.out.println("  Sintomas: " + sintomas);
        System.out.println("  Fecha de creacion: " + fechaCreacion);
        System.out.println("  Veterinario asignado: " + (veterinarioAsignado != null ? veterinarioAsignado.getNombre() : "Sin asignar"));
        if (diagnostico != null && !diagnostico.isEmpty())
        {
            System.out.println("  Diagnostico: " + diagnostico + " | Tratamiento: " + tratamiento);
        }
        System.out.println("  Escalado: " + (escalado ? "Si" : "No"));
    }

    public void mostrarBitacora()
    {
        mostrarResumen();
        bitacora.mostrarBitacora();
    }
}
